### ISSUE TITLE
Nullability and Requireness

### ISSUE THREAD

**Issue by cmdjulian** [creek-service/creek-json-schema#190]
**Is your feature request related to a problem? Please describe.**
I was wondering, as Jackson jsonSchema Generator supports nullability, if I can use it as well? The main problem arises from one of my kotlin data classes, which allows certain attributes to be null by default. The generated schema however says different.  
![image](https://github.com/creek-service/creek-json-schema/assets/33688601/cca5ee87-7b40-4813-89cd-9948a3c17b90)

It is closely related to requireness. I want to mark all attributes non-nullable and required by default, but only certain attributes I do wanna explicitly mark as nullable.

**Describe the solution you'd like**
A way to enable nullability transitively in the plugin

**Describe alternatives you've considered**
-

**Additional context**
-

Any help would be appreciated.

**Comment by cmdjulian** [comment]
I was also asking myself, if there is a way to make the plugin aware of any defaults provided by kotlin via its default args construct

**Comment by big-andy-coates** [comment]
Hi @cmdjulian,

I'm with you on this: I hate nulls!  Let me take a look.

**Comment by big-andy-coates** [comment]
> I was also asking myself, if there is a way to make the plugin aware of any defaults provided by kotlin via its default args construct

Can you move this out into its own issue, please?

**Comment by big-andy-coates** [comment]
> I want to mark all attributes non-nullable and required by default, but only certain attributes I do wanna explicitly mark as nullable.

I've pulled this question out into its own issue: https://github.com/creek-service/creek-json-schema/issues/204

**Comment by big-andy-coates** [comment]
> The main problem arises from one of my kotlin data classes, which allows certain attributes to be null by default. The generated schema however says different.

Taking a stab at answering this while waiting on your example code & schema...

The schemas generated by this project are opinionated, in that they don't allow explicit null values in JSON.  This is a deliberate design choice, made to encourage null-free coding practices.

Instead, what the plugin does support is optional fields, i.e. a field that may or may not be in the JSON. (Set using Jackson's `@JsonProperty(required = true|false)`).  This can be used in place of nullable fields in the schema if you ensure the field is only included in the JSON object if it has a non-null value. Jackson's `@JsonInclude(JsonInclude.Include.NON_EMPTY)` or `JsonMapper.serializationInclusion(JsonInclude.Include.NON_EMPTY)` can be used to ensure this.

**Comment by big-andy-coates** [comment]
Closing due to inactivity,.
